packages/web/src/features/users/UsersView.tsx

```diff
@@ -20,7 +20,7 @@ import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
 import SelectInput from '../../shared/components/form/SelectInput';
 
 export default function UsersView() {
-  const session = useSession();
+  const session = useAuth({ systemRoles: [SystemRole.Admin] });
```

Comment:
I wanted a way to restrict who can get to certain views based on their roles. This used to be `useSession` to get the current user, but I coopted it let you specify which roles can access, and navigate away if the current user doesn't pass. What do you think of this approach?

Reply:
So `useAuth` acts like a gatekeeper? I like it.